### PR TITLE
Read kernel defaults from a file instead of the kernel boot jar manifest.

### PR DESCRIPTION
**Pull request by kaczyns**
Currently, the only way for a product that extends Liberty to over-ride the default kernel or kernel extension, is for it to specify that kernel or extension as a system property.  It would be convenient to be able to specify extensions in a file instead, since that would eliminate the need to set the system property on every server instance.  Since to date all kernel extensions are platform-specific, we are introducing a platform-specific defaults over-ride file.  The file name is "lib/platform/defaults-*" where the asterisk is replaced by the os.name returned by the JVM.  If that file exists, we will read it, and it will over-ride what is specified in "lib/platform/defaults".

We also discovered that we were reading the defaults from the kernel boot jar manifest, instead of from the defaults file.  While this is fine, it makes more sense to read from the file since that can be updated more easily than changing the jar manifest.  We've changed how that works.

Resolves issue #425 
#build

### CODE REVIEW

**Comment by kaczyns** [comment]
#build

File: dev/com.ibm.ws.kernel.boot.core/src/com/ibm/ws/kernel/boot/internal/BootstrapDefaults.java [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+import java.util.Properties;
+
+import com.ibm.ws.kernel.boot.BootstrapConfig;
+
+/**
+ * Wrapper around the Liberty defaults file (in wlp/lib/platform/defaults).
+ */
+public class BootstrapDefaults {
+    /** Manifest header for the default kernel version */
+    static final String MANIFEST_KERNEL = "WebSphere-DefaultKernel";
```
Comment (tjwatson) [inline dev/com.ibm.ws.kernel.boot.core/src/com/ibm/ws/kernel/boot/internal/BootstrapDefaults.java]: While the property keys look like manifest headers we no longer read them as manifest headers.  Should update the javadoc comment to indicate this is a property key read out of the defaults file

File: dev/com.ibm.ws.kernel.boot.core/src/com/ibm/ws/kernel/boot/internal/KernelBootstrap.java [hunk 2]
```diff
@@ -135,12 +142,7 @@ public ReturnCode go() {
 
             // Find the bootstrap resources we need to launch the nested framework.
             // MAY THROW if these resources can not be found or read
-            KernelResolver resolver = new KernelResolver(bootProps.getInstallRoot(),
-                            bootProps.getWorkareaFile(KernelResolver.CACHE_FILE),
-                            bootManifest.getKernelDefinition(bootProps),
-                            bootManifest.getLogProviderDefinition(bootProps),
-                            bootManifest.getOSExtensionDefinition(bootProps),
-                            libertyBoot);
+            KernelResolver resolver = new KernelResolver(bootProps.getInstallRoot(), bootProps.getWorkareaFile(KernelResolver.CACHE_FILE), bootDefaults.getKernelDefinition(bootProps), bootDefaults.getLogProviderDefinition(bootProps), bootDefaults.getOSExtensionDefinition(bootProps), libertyBoot);
```
Comment (tjwatson) [inline dev/com.ibm.ws.kernel.boot.core/src/com/ibm/ws/kernel/boot/internal/KernelBootstrap.java]: Not sure what happened to the formatting here.  Perhaps just a glitch in the web UI for the PR.

**Comment by kaczyns** [comment]
The errors in the test build have been accounted for.
com.ibm.ws.collective.rest_fat.rw FAT tests

**Pull request merged**